### PR TITLE
[PW-1909] Write tests for PW-1027

### PR DESCRIPTION
**Pull request by prasanthkrajan**
# Refer to the issue id if any. Include the link to the issue. E.g:
[Write tests for PW-1027](https://perxtechnologies.atlassian.net/browse/PW-1909)

# Describe the changes made. What does this PR changes that might be critical. If any critical decisions have been made, make sure you explain the rationale for these decisions.

- Added tests for template rendering in Comm Service

# How does the implementation addresses the problem

### CODE REVIEW

File: services/comm/app/models/template.rb [hunk 1]
```diff
@@ -11,6 +11,8 @@ def initialize_properties
 
   # See: https://www.stuartellis.name/articles/erb/
   def render(user, campaign)
+    return content if user.nil? || campaign.nil?
```
Comment (rpbaltazar) [inline services/comm/app/models/template.rb]: This is not necessarily true. What if we don't want to refer to the campaign in the sms? Or user?
I think that, at most, one could argue that this is useless if both are nil

**Comment by rpbaltazar** [comment]
@prasanthkrajan still needs to be cleaned up

**Comment by prasanthkrajan** [comment]
> @prasanthkrajan still needs to be cleaned up

Yes, still WIP. not ready yet. Will update once ready. @rpbaltazar

File: services/comm/spec/models/template_spec.rb [hunk 2]
```diff
@@ -5,8 +5,44 @@
 RSpec.describe Template, type: :model do
   include_examples 'application record concern' do
     let(:tenant) { Tenant.first }
-    let!(:subject) { create(factory_name) }
+    let!(:first_name) { 'Jim' }
```
Comment (rpbaltazar) [inline services/comm/spec/models/template_spec.rb]: This does not need to live here. We're including the examples 'application record concern' which are shared examples. None of the tests in that shared context need any of these records to be created.

I believe you might need some of them for the render context, so then these should be created in the render context and not somewhere else randomly.

File: services/comm/spec/models/template_spec.rb [hunk 3]
```diff
@@ -5,8 +5,44 @@
...
+    let!(:title) { 'Mr.' }
```
Comment (rpbaltazar) [inline services/comm/spec/models/template_spec.rb]: Also, as a general rule of thumb, `let` is preferred in regards to `let!`
The main difference is that `let` is lazily evaluated (evaluated when invoked), while `let!` is evaluated at execution time. This also means that if during some test setup you use `let` and you never reference that variable, then the code is never executed which can save you some precious seconds in the long run.

**Pull request merged**